Pyfluent Cheat Sheet & updated respective changes (#13)
pyansys/pyansys-cheat-sheet: cheat_sheets/pyfluent_cheat_sheet/pyfluent_cheat_sheet.tex

\documentclass[9pt,landscape]{article}
\usepackage{{./../static/style}}
\pdfinfo{
  /Title (PyAnsys Cheat Sheet)
  /Creator (TeX)
  /Producer (pdfTeX 1.40.0)
  /Author (Ansys)
  /Subject (PyAnsys)
  /Keywords (PyAnsys, Cheat sheet, template)}

\begin{document}
\raggedright
\footnotesize

% Add the title of cheat sheet here
% ----------------------------------------

\begin{center}
     \Huge{\textbf{Cheat sheet for PyFluent}} \\
\end{center}
\begin{center}
     \Large{\textbf{Solver Settings Object Interface}} \\
\end{center}

\AddToShipoutPicture*
  {\put(670,577.5){\includegraphics[height = 1.2cm]{ansys.png}}}
\AddToShipoutPictureBG*{\includegraphics[width=\paperwidth]{bground.png}}
\vspace{-0.15cm}
\noindent\makebox[\linewidth]{\rule{\paperwidth}{2pt}}

\begin{multicols}{3}
\setlength{\premulticols}{1pt}
\setlength{\postmulticols}{1pt}
\setlength{\multicolsep}{1pt}
\setlength{\columnsep}{2pt}

% session starts here. 
% First colomn
% --------------------------------------------------------------------------------
\vfill
\section{\includegraphics[height=\fontcharht\font`\S]{slash.png} Launch Fluent locally}

> Following method to start Fluent from Python in gRPC mode.\\
> This code starts Fluent in the background so that commands can be sent to Fluent from the Python interpreter.

Reviewer: ```suggestion
The following method is used to start Fluent from Python in gRPC mode.\\
This code starts Fluent in the background so that commands can be sent to Fluent from the Python interpreter.
```

\begin{lstlisting}[language=Python]
import ansys.fluent.core as pyfluent
solver = pyfluent.launch_fluent(mode = "solver", show_gui = True)
\end{lstlisting}

The solver object contains attributes such as \textbf{file}, \textbf{setup}, \textbf{solution}, and \textbf{results}, 
which are also instances of settings objects. 

% row 2 col 1
\section{\includegraphics[height=\fontcharht\font`\S]{slash.png} Import mesh in launched session}

> The following examples show reading the available mesh file in Fluent Session.

Reviewer: ```suggestion
The following examples show how to read the available mesh file in Fluent Session.
```

\begin{lstlisting}[language=Python]
import_filename = 'example_file.msh.h5'
solver.file.read(file_type="case", file_name=import_filename)
\end{lstlisting}

> There are other specific APIs available for reading case file and
> reading case-data file. 

Reviewer: ```suggestion
There are other specific APIs available for reading case files and
reading case-data files. 
```

\begin{lstlisting}[language=Python]

>   # e.g., read_case(), read_case_data()
> 
>   import_filename = 'example_file.cas.h5'
>   solver.file.read_case(file_type="case", file_name=import_filename)  

Reviewer 2: ```suggestion
# e.g., read_case(), read_case_data()

import_filename = 'example_file.cas.h5'
solver.file.read_case(file_type="case", file_name=import_filename)  
```

\end{lstlisting}

% row 3 col 1
\section{\includegraphics[height=\fontcharht\font`\S]{slash.png} Enable heat transfer physics}

> Following examples show to enable heat transfer by activating the energy equation.

Reviewer: ```suggestion
The following examples show how to enable heat transfer by activating the energy equation.
```

\begin{lstlisting}[language=Python]
solver.setup.models.energy.enabled = True
\end{lstlisting}

{\includegraphics[height=\fontcharht\font`\S]{slash.png} Accessing the \textbf{object state} with using \textbf{pprint}

\begin{lstlisting}[language=Python]

>   >>> from pprint import pprint
>   >>> pprint (solver.setup.models.energy())
>   {'enabled': True,
>   'inlet_diffusion': True,
>   'kinetic_energy': False,
>   'pressure_work': False,
>   'viscous_dissipation': False}  

Reviewer 2: ```suggestion
>>> from pprint import pprint
>>> pprint(solver.setup.models.energy())
  {'enabled': True,
  'inlet_diffusion': True,
  'kinetic_energy': False,
  'pressure_work': False,
  'viscous_dissipation': False}  
```

\end{lstlisting}

% Second column
% --------------------------------------------------------------------------------
% row 1 col 2
\vfill
\section{\includegraphics[height=\fontcharht\font`\S]{slash.png}  Define materials}
This example shows how you use Solver settings objects to define materials.
\begin{lstlisting}[language=Python]

> 
> solver.setup.materials.copy_database_material_by_name(type="fluid", name="water-liquid")
> solver.setup.cell_zone_conditions.fluid["elbow-fluid"].material = "water-liquid"
> 
> \end{lstlisting} 

Reviewer: ```suggestion
solver.setup.materials.copy_database_material_by_name(type="fluid", name="water-liquid")
solver.setup.cell_zone_conditions.fluid["elbow-fluid"].material = "water-liquid"
\end{lstlisting} 
```

% row 2 col 2
\section{\includegraphics[height=\fontcharht\font`\S]{slash.png}  Define boundary conditions}

The examples in this section show how you use Solver settings objects to define boundary conditions.

\begin{lstlisting}[language=Python]

> 

Reviewer: ```suggestion
```

solver.setup.boundary_conditions.velocity_inlet["cold-inlet"].vmag = {
    "option": "constant or expression",
    "constant": 0.4,
}
solver.setup.boundary_conditions.velocity_inlet[
    "cold-inlet"
].ke_spec = "Intensity and Hydraulic Diameter"
solver.setup.boundary_conditions.velocity_inlet["cold-inlet"].turb_intensity = 5
solver.setup.boundary_conditions.velocity_inlet[
    "cold-inlet"
].turb_hydraulic_diam = "4 [in]"
solver.setup.boundary_conditions.velocity_inlet["cold-inlet"].t = {
    "option": "constant or expression",
    "constant": 293.15,
}
\end{lstlisting} 

\vfill
{\includegraphics[height=\fontcharht\font`\S]{slash.png}  Modify \textbf{cell zone} conditions} \\
The examples in this section show how you use Solver settings objects to modify \textbf{cell zone} conditions.
\begin{lstlisting}[language=Python]

> 

Reviewer: ```suggestion
```

# Enabling Laminar Zone

solver.setup.cell_zone_conditions.fluid["elbow-fluid"] = {"laminar": True}  

> 
> 

Reviewer: ```suggestion
```

\end{lstlisting}
\vfill

% Third column
% --------------------------------------------------------------------------------
\vfill
% row 1 col 3
\section{\includegraphics[height=\fontcharht\font`\S]{slash.png}  Apply solution settings}
PyFluent allows you to use Solver settings objects to apply
solution settings, initialize, and solve.

\begin{lstlisting}[language=Python]

> 

Reviewer: ```suggestion
```

solver.solution.initialization.hybrid_initialize()
solver.solution.run_calculation.iterate(number_of_iterations=150)

> 

Reviewer: ```suggestion
```

\end{lstlisting}

% row 2 col 3
\section{\includegraphics[height=\fontcharht\font`\S]{slash.png}  Post-Processing}

> PyFluent allows you yo Post-Process data with \textbf{results} object. Following example show you to create and display contour on plane

Reviewer: ```suggestion
PyFluent allows you to post-process data with \textbf{results} object. The following example shows you how to create and display contours on a plane.
```
